### PR TITLE
Move env vars to script, add build scan wrapper

### PR DESCRIPTION
**Pull request by alpar-t**
Having the env vars being set up in the build script makes it easier to
reason about.
This PR also adds the gradle scan wrapper back as support for it was
added t o jjbb.

### CODE REVIEW

File: .ci/build.sh [hunk 1]
```diff
@@ -11,8 +22,8 @@ if [ -f /proc/cpuinfo ] ; then
 else
    if [[ "$OSTYPE" == "darwin"* ]]; then
       MAX_WORKERS=`sysctl -n hw.physicalcpu | sed 's/\s\+//g'`
```
Comment (rjernst) [inline .ci/build.sh]: This line seems to be irrelevant since we hardcode max workers to 2 below?

**Comment by atorok** [comment]
It is, but I'm not convinced we need to hard code so I didn't want to
remove it. I'll add a comment to explain
